Raven.Client.Lightweight/Changes/RemoteChangesClientBase.cs

```diff
@@ -201,83 +213,149 @@ private bool DisposeHeartbeatTimer()
             }
         }
 
-        private Task lastSendTask;
-
-        protected Task Send(string command, string value)
+        async Task Worker()
         {
-            lock (this)
+            while (tokenSource.IsCancellationRequested == false)
             {
-                logger.Info("Sending command {0} - {1} to {2} with id {3}", command, value, url, id);
-                var sendTask = lastSendTask;
-                if (sendTask != null)
+                try
                 {
-                    return sendTask.ContinueWith(_ =>
-                    {
-                        Send(command, value);
-                    });
-                }
+                    await syncSource.Task.ConfigureAwait(false);
+                    syncSource = TaskCompletionSourceFactory.Create<bool>();
 
-                try
+                    await WorkUntilEmpty().ConfigureAwait(false);
+                }
+                catch (TaskCanceledException)
                 {
-                    var sendUrl = url + "/changes/config?id=" + id + "&command=" + command;
-                    if (string.IsNullOrEmpty(value) == false)
-                        sendUrl += "&value=" + Uri.EscapeUriString(value);
+                    // Swallowing the task cancellation in case we are stopping work.
+                }
+            }
 
-                    var requestParams = new CreateHttpJsonRequestParams(null, sendUrl, HttpMethods.Get, credentials, Conventions)
-                    {
-                        AvoidCachingRequest = true
-                    };
-                    var request = jsonRequestFactory.CreateHttpJsonRequest(requestParams);
-                    lastSendTask = request.ExecuteRequestAsync().ObserveException();
+            // drain
+            await WorkUntilEmpty().ConfigureAwait(false);
```

Comment:
Need try catch for `TaskCancelledException` here, no?

Reply:
`WorkUntilEmpty` doesn't observe the the token or the done task. So I don't think it is needed